pygfx/utils/text/_shaper.py

```diff
@@ -73,8 +78,12 @@ def check_lifetimes(self):
         # it slower already for 100 items, which is way more realistic
         # font count. So let's keep things simple :)
         pretty_old = time.time() - self._ref_lifetime
-        to_remove = {key for key, lt in self._lifetimes.items() if lt < pretty_old}
-        for key in to_remove:
+        to_remove = sorted(
+            ((key, lt) for key, lt in self._lifetimes.items() if lt < pretty_old),
+            key=lambda x: x[1],
+        )
+        items_to_remove = max(len(self._cache) - self._minimum_items, 0)
```

Comment:
I'm not sure I follow the logic here. The `._minimum_items` applies to the subset of items that are older than `pretty_old`.  So I think this should be something like:

```py
        n_pretty_new = len(self._cache) - len(to_remove)
        items_to_remove = max(n_pretty_new - self._minimum_items, 0)
```

Reply:
I don't think it works out.

consider:

* len(self._cache) = 10
* 9 items are "too old"
* minimum_items = 5
* We thus want at the end to have 5 items in our cache limited by `minimum_items`

thus
* n_pretty_new = 10 - 9 = 1
* items_to_remove = max(1 - 5, 0) = max(-4, 0)
* so you remove nothing???

I renamed the varaible to `max_items_to_remove`

The slicing operator effectively does

```
items_to_remove = min(max_items_to_remove, len(pretty_old))
```

Reply:
My brain was thinking in the wrong direction. Now I get it. The `max_items_to_remove` is simply how many items can we remove, let's removed this many items, starting with the oldest. The renaming helps. I applied another rename to help further.